### DISCUSSION TITLE
Is the arena allocation safe from invisible "safe unsafety"?

### DISCUSSION THREAD

**Discussion by lahwran** [leptos-rs/leptos#965] [Q&A] [answered]
Hey! I'm not a heavy rust programmer, just trying to pick a gui library and shopping (don't tell anyone, but I'm maybe having more fun than necessary shopping). I'd like to use leptos and not have to fix or identify particularly esoteric bugs. I like the design overall, but can I trust that the arena allocation isn't going to have, eg, memory leaks or allocation conflicts, that the borrow checker can't see because they're not real rust ownership? What sort of protection against that is designed into leptos?

**Comment by gbj** [comment, marked as answer]
> Hey! I'm not a heavy rust programmer, just trying to pick a gui library and shopping (don't tell anyone, but I'm maybe having more fun than necessary shopping). 

That's the most fun part!

> I'd like to use leptos and not have to fix or identify particularly esoteric bugs. I like the design overall, but can I trust that the arena allocation isn't going to have, eg, memory leaks or allocation conflicts, that the borrow checker can't see because they're not real rust ownership? What sort of protection against that is designed into leptos?

Leptos itself doesn't include any unsafe code; it's built on the safe interface of slotmap. 

The worst memory issues that can happen are memory leaks, usually due to some kind of failure on my part to think ahead and manually create/dispose of reactive scopes in some of the control flow components built into the framework. These happen occasionally and are usually pretty straightforward to resolve when reported. 

Otherwise I'd say we don't really run into these issues very much. The framework has some bugs of course but usually not on this level.
